macros/src/codegen.rs

```diff
@@ -572,3 +534,14 @@ pub fn generate_code(sm: &ParsedStateMachine) -> proc_macro2::TokenStream {
         }
     }
 }
+
+impl ToTokens for Lifetimes {
+    fn to_tokens(&self, tokens: &mut TokenStream) {
+        if self.is_empty() {
+            return;
+        }
+
+        let lifetimes = self.as_slice();
+        tokens.extend(quote! { #(#lifetimes),* ,});
+    }
+}
```

Comment:
This should go into the `lifetime.rs` file if it's an explicit impl on that type to keep relevant code together IMO. Sorry for the mixed messaging.